### ISSUE TITLE
early_stop_patience_steps not working with AutoTSMixer

### ISSUE THREAD

**Issue by almeidajulianojm** [Nixtla/neuralforecast#1076]
### What happened + What you expected to happen

I'm defining the config of the AutoTSMixer model using the following dictionary:

tsmixer_config = {
        "input_size": tune.choice([4, 8, 16, 32]), 
        "max_steps": tune.choice([200, 500, 1000]), 
        "val_check_steps": 100, 
        "early_stop_patience_steps": 5, 
        "learning_rate": tune.loguniform(1e-4, 1e-2), 
        "n_block": tune.choice([1, 2, 4, 6, 8]), 
        "dropout": tune.uniform(0.0, 0.99), 
        "ff_dim": tune.choice([32, 64, 128]), 
        "scaler_type": 'identity',       
        }

However, trying to use _early_stop_patience_steps_ in this configuration gives the following error:

> RuntimeError: Early stopping conditioned on metric `ptl/val_loss` which is not available. Pass in or modify your `EarlyStopping` callback to use any of the following: `train_loss`, `train_loss_step`, `train_loss_epoch`

This is a similar bug as with the Issue #688. However, I'm passing val_size and test_size to the cross_validation method as required. 

Removing the line ""early_stop_patience_steps": 5," solves the problem as it uses the default configuration.

Please let me know if I need to provide any additional information, this is my first issue :)

### Versions / Dependencies

Using Debian GNU/Linux 12 with Python 3.11.2 and the following libraries:

absl-py==2.1.0
aiohttp==3.9.5
aiosignal==1.3.1
alembic==1.13.1
altair==5.3.0
asttokens==2.4.1
astunparse==1.6.3
attrs==23.2.0
Babel==2.15.0
bcrypt==4.1.3
beautifulsoup4==4.12.3
blinker==1.8.2
branca==0.7.2
bs4==0.0.2
cachetools==5.3.3
certifi==2024.2.2
charset-normalizer==3.3.2
click==8.1.7
click-plugins==1.1.1
cligj==0.7.2
cloudpickle==3.0.0
cmdstanpy==1.2.2
colorama==0.4.6
colorlog==6.8.2
comm==0.2.2
contourpy==1.2.1
coreforecast==0.0.10
cycler==0.12.1
Cython==3.0.10
debugpy==1.8.2
decorator==5.1.1
entrypoints==0.4
executing==2.0.1
extra-streamlit-components==0.1.71
Faker==25.1.0
fastjsonschema==2.20.0
favicon==0.7.0
filelock==3.15.3
fiona==1.9.6
flatbuffers==24.3.25
folium==0.16.0
fonttools==4.51.0
frozenlist==1.4.1
fsspec==2024.6.0
future==1.0.0
gast==0.5.4
geopandas==0.14.4
gitdb==4.0.11
GitPython==3.1.43
google-pasta==0.2.0
greenlet==3.0.3
grpcio==1.63.0
h5py==3.11.0
holidays==0.48
htbuilder==0.6.2
hydralit-components==1.0.10
hyperopt==0.2.7
idna==3.7
importlib_metadata==7.1.0
importlib_resources==6.4.0
ipykernel==6.29.5
ipython==8.26.0
ipywidgets==8.1.3
jedi==0.19.1
Jinja2==3.1.4
joblib==1.4.2
jsonschema==4.22.0
jsonschema-specifications==2023.12.1
jupyter_client==8.6.2
jupyter_core==5.7.2
jupyterlab_widgets==3.0.11
keras==3.3.3
kiwisolver==1.4.5
libclang==18.1.1
lightning-utilities==0.11.2
lxml==5.2.1
Mako==1.3.5
Markdown==3.6
markdown-it-py==3.0.0
markdownlit==0.0.7
MarkupSafe==2.1.5
matplotlib==3.9.1
matplotlib-inline==0.1.7
mdurl==0.1.2
ml-dtypes==0.3.2
more-itertools==10.2.0
mpmath==1.3.0
msgpack==1.0.8
multidict==6.0.5
namex==0.0.8
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.3
neuralforecast==1.7.2
numpy==1.26.4
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.20.5
nvidia-nvjitlink-cu12==12.5.40
nvidia-nvtx-cu12==12.1.105
opt-einsum==3.3.0
optree==0.11.0
optuna==3.6.1
packaging==24.0
pandas==2.2.2
parso==0.8.4
patsy==0.5.6
pexpect==4.9.0
pillow==10.3.0
platformdirs==4.2.2
plotly==5.22.0
plotly-express==0.4.1
pmdarima==2.0.4
prettytable==3.10.0
prometheus_client==0.20.0
prompt_toolkit==3.0.47
prophet==1.1.5
protobuf==4.25.3
psutil==6.0.0
ptyprocess==0.7.0
pure-eval==0.2.2
py4j==0.10.9.7
pyarrow==16.0.0
pydeck==0.9.0
pyecharts==2.0.5
Pygments==2.18.0
PyJWT==2.8.0
pymdown-extensions==10.8.1
Pympler==1.0.1
pyparsing==3.1.2
pyproj==3.6.1
PyQt6==6.7.0
PyQt6-Qt6==6.7.2
PyQt6-sip==13.6.0
python-dateutil==2.9.0.post0
python-decouple==3.8
pytorch-lightning==2.3.0
pytz==2024.1
pytz-deprecation-shim==0.1.0.post0
PyYAML==6.0.1
pyzmq==26.0.3
ray==2.24.0
referencing==0.35.1
requests==2.31.0
rich==13.7.1
rpds-py==0.18.1
schedule==1.2.1
scikit-learn==1.4.2
scipy==1.13.0
seaborn==0.13.2
shapely==2.0.4
simplejson==3.19.2
six==1.16.0
smmap==5.0.1
soupsieve==2.5
SQLAlchemy==2.0.31
st-annotated-text==4.0.1
st-pages==0.4.5
stack-data==0.6.3
stanio==0.5.0
statsmodels==0.14.2
streamlit==1.34.0
streamlit-aggrid==0.3.4.post3
streamlit-antd-components==0.3.2
streamlit-authenticator==0.2.2
streamlit-camera-input-live==0.2.0
streamlit-card==1.0.0
streamlit-echarts==0.4.0
streamlit-embedcode==0.1.2
streamlit-extras==0.4.2
streamlit-faker==0.0.3
streamlit-image-coordinates==0.1.6
streamlit-keyup==0.2.4
streamlit-server-state==0.17.1
streamlit-toggle-switch==1.0.2
streamlit-vertical-slider==2.5.5
sympy==1.12.1
tenacity==8.3.0
tensorboard==2.16.2
tensorboard-data-server==0.7.2
tensorboardX==2.6.2.2
tensorflow==2.16.1
tensorflow-io-gcs-filesystem==0.31.0
termcolor==2.4.0
threadpoolctl==3.5.0
toml==0.10.2
toolz==0.12.1
torch==2.3.1
torchmetrics==1.4.0.post0
tornado==6.4
tqdm==4.66.4
traitlets==5.14.3
triton==2.3.1
typing_extensions==4.11.0
tzdata==2024.1
tzlocal==5.2
urllib3==2.2.1
utilsforecast==0.1.11
validators==0.28.1
watchdog==4.0.0
wcwidth==0.2.13
Werkzeug==3.0.3
widgetsnbextension==4.0.11
wrapt==1.16.0
xyzservices==2024.4.0
yarl==1.9.4
zipp==3.18.1

### Reproduction script

```
  input_size = 32
  n_samples = 1
  
  n_series = df.unique_id.nunique()
  
  dataframe_size = len(df.ds.unique())
  val_size = int(0.1*dataframe_size)
  test_size = int(0.1*dataframe_size)
  
  tsmixer_config = {
      "input_size": tune.choice([4, 8, 16, 32]),          # Size of input window
      "max_steps": tune.choice([200, 500, 1000]),         # Number of training iterations
      "val_check_steps": 100,                             # Compute validation every x steps
      "early_stop_patience_steps": 5,                     # Early stopping steps
      "learning_rate": tune.loguniform(1e-4, 1e-2),       # Initial Learning rate
      "n_block": tune.choice([1, 2, 4, 6, 8]),            # Number of mixing layers
      "dropout": tune.uniform(0.0, 0.99),                 # Dropout
      "ff_dim": tune.choice([32, 64, 128]),               # Dimension of the feature linear layer
      "scaler_type": 'identity',       
      }
  
  tsmixerx_config = tsmixer_config.copy()
  tsmixerx_config['futr_exog_list'] = [
      'price', 
      'buy_dollar', 
      'ipca', 
      'selic', 
      '3000 SEMI ACABADOS', 
      '4000 PARTES E PECAS', 
      '5000 CORR TOP & FLEX', 
      '7000 CORR MODULARES'
      ]
  
  models = [
      AutoTSMixer(
          h=forecasting_horizon, 
          n_series=n_series,  
          config=tsmixer_config, 
          num_samples=n_samples, 
          search_alg=HyperOptSearch(), 
          backend='ray',
          loss=RMSE(),
          valid_loss=RMSE(),
          ),
      # AutoTSMixerx(
      #     h=forecasting_horizon, 
      #     n_series=n_series, 
      #     config=tsmixerx_config, 
      #     num_samples=n_samples, 
      #     search_alg=HyperOptSearch(), 
      #     backend='ray',
      #     loss=RMSE(), 
      #     valid_loss=RMSE(),
      #     )                                        
          ]
  
  nf = NeuralForecast(models=models, freq=freq)
  y_pred = nf.cross_validation(df=df, val_size=val_size, test_size=test_size, n_windows=None)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by jmoralez** [comment]
Hey @almeidajulianojm, thanks for the detailed report. Can you try upgrading to 1.7.3? I believe this was fixed by #1031

**Comment by almeidajulianojm** [comment]
Thank you for the super fast reply. Upgrading to 1.7.3 solved the problem. Thank you for your help :+1:
